metrics/tarantool/event_loop.lua

```diff
@@ -0,0 +1,43 @@
+local clock = require('clock')
+local fiber = require('fiber')
+local metrics = require('metrics')
+
+local collectors_list = {}
+
+local function monitor()
+    fiber.self():name("tx_speed")
+
+    local max = -math.huge
+
+    while true do
+        local start = clock.monotonic64()
+        fiber.yield()
```

Comment:
Firstly, monotonic64 doesn't show nearly realtime. It may produce times which a not related to real wallclock.

It seems that ev_time (fiber.time()) is actually uses simple clock_gettime
https://github.com/enki/libev/blob/master/ev.c#L1854

Secondly, it is too bad to run this loop without fiber.sleep(interval), because now this fiber will be constantly scheduled in each ev_once.

It is good to notice, that diff will be measured at the beginning of the single run, while start is measured nearly at any point in RUN (but in this code also in the beginning of the loop).

I suggest you to write something like this:
```lua
fiber.sleep(interval) -- sleep interval, and this will guarantee us that next line will be executed nearly at the beginning of the loop (after some timers)
local wall_1, ev_1 = clock.realtime64(), fiber.time64()*1e3
fiber.sleep(0) -- reset watcher to EV, to reach next loop. Actually EV has ev_postpone, but we do not have such API in Lua.
local wall_2, ev_2 = clock.realtime64(), fiber.time64()*1e3

-- Do your mesurments: you will have 3 intervals:
-- (if everything is ok with ntp):
-- ev_1 <= wall_1 <= ev_2 <= wall_2
-- ev_loop_time is ev_2 - ev_1 (it is mesured by the EV itself at the beginning of the loop)
-- wall_2 - wall_1 -- is actual sleep of the fiber. It does mean not that much actually
-- wall_2 - ev_2 -- is lag from start of this run of loop (if it is large, then we might have some long code in timers)
-- ev_2 - wall_1 -- is lag of "epilog" of the loop. If it is large then exists fibers which are running long while having fiber.sleep(timeout > 0) or long requests
```

Read more info about my experiments: https://gist.github.com/ochaton/6bc9a7030df7b4e7c63703599a5b2f96

Also you have to provide mechanism to stop your monitoring fiber and maybe get collector before while-loop